src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs

```diff
@@ -686,38 +691,44 @@ public async Task<VersionStamp> GetSemanticVersionAsync(CancellationToken cancel
                 analyzerConfigSet ?? _lazyAnalyzerConfigOptions);
         }
 
-        private ProjectInfo.ProjectAttributes Attributes
+        internal ProjectInfo.ProjectAttributes Attributes
             => ProjectInfo.Attributes;
 
-        private ProjectState WithAttributes(ProjectInfo.ProjectAttributes attributes)
+        private ProjectState WithRawAttributes(ProjectInfo.ProjectAttributes attributes)
             => With(projectInfo: ProjectInfo.With(attributes: attributes));
 
+        public ProjectState WithAttributes(ProjectInfo.ProjectAttributes attributes)
+            => (attributes == Attributes) ? this : WithRawAttributes(attributes.With(version: Version.GetNewerVersion()));
```

Comment:
Similar to other comments: this is tricky since if anything is changing here, then it's hard to know how this needs to update DocumentStates.